### ISSUE TITLE
`random.choice` on multidimensional arrays

### ISSUE THREAD

**Issue by juliuskunze** [google/jax#8158]
Can we model `jax.random.choice` after the newer [`numpy.random.Generator.choice`](https://numpy.org/doc/stable/reference/random/generated/numpy.random.Generator.choice.html) instead of [`numpy.random.choice`](https://numpy.org/doc/stable/reference/random/generated/numpy.random.choice.html)? See [numpy discussion](https://github.com/numpy/numpy/issues/2724) for reference.

This requires adding an `axis=0` parameter which would generalize the current implementation and keep it backward-compatible.
